lib/order.rb

```diff
@@ -0,0 +1,49 @@
+require_relative 'menu'
+require 'date'
+require 'twilio-ruby'
+
+class Order
+
+  attr_reader :final_order
+
+  # put your own credentials here
+  ACCOUNT_SID = 'ACbb9d79e8b89cc29c5363dde46532fcee' 
+  AUTH_TOKEN = '485416945d5c976283c4a652ad691468' 
+  # set up a client to talk to the Twilio REST API
+  CLIENT = Twilio::REST::Client.new(ACCOUNT_SID, AUTH_TOKEN)
+
+  def initialize(menu)
+    @menu = menu
+    @final_order = []
+  end
+
+  def add_order(dish_name, quantity)
+    fail "Sorry, #{dish_name} is not available"\
+     unless @menu.dish?(dish_name) 
+    dish_obj = @menu.get_dish_obj(dish_name)
+    @final_order.push({ dish: dish_obj, quantity: quantity })
+  end
+
```

Comment:
How about something like this instead of lines 28 to 31?

```
item_totals = @final_order.map { | item | item[:dish].price * item[:quantity] }
grand_total = item_totals.sum
```

I think I would do something like that but I am curious to know whether you think my solution is clearer or not.  There might also be another solution using a single method.  In fact, I just found a way of doing that but it's not very clear:

`grand_total = @final_order.reduce(0) { |sum, item| sum + ( item[:dish].price )}`

What do you think about these options?